### PR TITLE
Add support for Apple login

### PR DESCRIPTION
**Pull request by tiagojsag**

### CODE REVIEW

File: app/src/plugins/sd-ct-oauth-plugin/auth.router.js [hunk 1]
```diff
@@ -865,6 +931,10 @@ module.exports = (plugin, connection, generalConfig) => {
     ApiRouter.get('/twitter', setCallbackUrl, API.twitter);
     ApiRouter.get('/twitter/callback', API.twitterCallback, API.updateApplications);
 
+    ApiRouter.get('/apple', setCallbackUrl, API.apple);
+    ApiRouter.post('/apple/callback', API.appleCallback, API.updateApplications);
+    ApiRouter.get('/apple/token', API.appleToken, API.generateJWT);
```
Comment (ikas) [inline app/src/plugins/sd-ct-oauth-plugin/auth.router.js:936]: is `API.generateJWT` needed? looks like there's a token being generated already at the end of the `API.appleToken` method

File: app/src/plugins/sd-ct-oauth-plugin/services/passport.service.js [hunk 2]
```diff
@@ -4,14 +4,53 @@ const { BasicStrategy } = require('passport-http');
...
+            id: user._id,
+            provider: user.provider,
+            providerId: user.providerId,
+            role: user.role,
+            createdAt: user.createdAt,
+            extraUserData: user.extraUserData,
+            name: user.name,
+            photo: user.photo,
+            email: user.email
+        });
```
Comment (ikas) [inline app/src/plugins/sd-ct-oauth-plugin/services/passport.service.js:51]: :thinking: ?

```suggestion
        done(null, user.toJSON());
```

File: app/test/e2e/ct-oauth/ct-oauth-apple.spec.js [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+                token_type: 'Bearer',
+                expires_in: 3600,
+                refresh_token: 'r1350f8480d5c45d3a74f3aaa7c0d979e.0.mzvy.J9iC84ON-bQEcEDspOa17w',
+                id_token: 'eyJraWQiOiJlWGF1bm1MIiwiYWxnIjoiUlMyNTYifQ.eyJpc3MiOiJodHRwczovL2FwcGxlaWQuYXBwbGUuY29tIiwiYXVkIjoib3JnLnJlc291cmNld2F0Y2guYXBpLmRldi5hdXRoIiwiZXhwIjoxNjAzOTYwOTY3LCJpYXQiOjE2MDM4NzQ1NjcsInN1YiI6IjAwMDk1OC5hNDU1MGE4ODA0Mjg0ODg2YTViNTExNmExYzAzNTFhZi4xNDI1IiwiYXRfaGFzaCI6InhWNkJBN2xZVU8takt6QTdHdGhrRXciLCJlbWFpbCI6ImRqOGU5OWczNG5AcHJpdmF0ZXJlbGF5LmFwcGxlaWQuY29tIiwiZW1haWxfdmVyaWZpZWQiOiJ0cnVlIiwiaXNfcHJpdmF0ZV9lbWFpbCI6InRydWUiLCJhdXRoX3RpbWUiOjE2MDM4NzQ1MDUsIm5vbmNlX3N1cHBvcnRlZCI6dHJ1ZX0.l_AYOAArJf1suJ3218CndgD4QVwoPklUEcHBN6ZSAXOlKRCDWNQgx7bCXPMZzZ6W2E4SYV2Hx53z0yJXeW_i-CeDyGpInWTihKywq22lN7DIsTfzyLQsgA2ShPf1JGiJ7953fm-QkBvkISd0SmgUDHWRtVKtFEAtaoEdOlnTR7-RFUV_APbpsz_vWzzJkwy-smyfKeh2CImfWoZHnzLXEUCcCpbfynx6k8MwUR2iVoUP6UPbwNoO7Pc6wxrwQYb8GSyGGW953A1BJhfWSwPFP8aCukyuHA5YEG329QTUphd9tpb92M27hnqK_GPJqBQhy-Ub7rehOiPNQZ3eoNTwpg'
+            });
+
+        await requester
+            .get(`/auth`);
```
Comment (ikas) [inline app/test/e2e/ct-oauth/ct-oauth-apple.spec.js:95]: not needed, I think

File: k8s/dev/cron.deployment.yaml [hunk 4]
```diff
@@ -72,6 +72,26 @@ spec:
                 secretKeyRef:
                   key: JWT_SECRET
                   name: control-tower
+            - name: APPLE_KEY_ID
```
Comment (ikas) [inline k8s/dev/cron.deployment.yaml:75]: do cron deployments need these secrets?

**Review by ikas — APPROVED** [review]
A few minor comments, LGTM otherwise :ok_hand:

File: app/src/plugins/sd-ct-oauth-plugin/auth.router.js (hunk 1 above)
Comment (tiagojsag) [reply app/src/plugins/sd-ct-oauth-plugin/auth.router.js:936]: good catch - I actually want to keep the call to generateJWT, and make the custom code match that.

File: app/src/plugins/sd-ct-oauth-plugin/services/passport.service.js (hunk 2 above)
Comment (tiagojsag) [reply app/src/plugins/sd-ct-oauth-plugin/services/passport.service.js:51]: my approach is more explicit as to what gets serialized and what doesn't. Typically this is preferable, as it requires explicit control over what gets passed out - for example, if you add a new field to the model, you need to explicitly add it here, and by default its not exposed. So if that field happens to be sensitive, by default, it stays hidden.

In the particular context of a user, this is even more important, as just blindly serializing data could expose passwords and salts, and that's halfway into getting attacked.

File: k8s/dev/cron.deployment.yaml (hunk 4 above)
Comment (tiagojsag) [reply k8s/dev/cron.deployment.yaml:75]: we probably dont

**Pull request merged**